precode_test.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMainHandlerWhenStatusOkAndBodyNotEmpty(t *testing.T) {
+    req := httptest.NewRequest("GET", "/cafe?count=10&city=moscow", nil)
```

Comment:
Подразумевается, что тут должно быть нормальное количество, которое точно есть. Иначе тест не отличается от следующего
```suggestion
    req := httptest.NewRequest("GET", "/cafe?count=2&city=moscow", nil)
```